text/XXX-flatten-feature.md

```diff
@@ -0,0 +1,325 @@
+
+Will group the given buildpacks into one layer and keep the other ones as single layers buildpacks, the result will be:
+
+```mermaid
+classDiagram
+    class Layer1 {
+      CB1
+    }
+    class Layer2 {
+      G1 
+    }
+    class Layer3 {
+      BP1
+      BP2
+    }
+    class Layer4 {
+      G2
+    }
+    class Layer5 {
+      BP3
+      BP4
+    }
+```
+
+
+$$ 
+ total = \text{5 layers}
+$$
+
+
+# Migration
+[migration]: #migration
+
+The current [distribution spec](https://github.com/buildpacks/spec/blob/main/distribution.md#buildpackage) defines:
+
+```
+Each buildpack layer blob MUST contain a single buildpack at the following file path:
+
+/cnb/buildpacks/<buildpack ID>/<buildpack version>/
+```
+
+A Buildpackage flattened with this new feature would not be consumable by older platform implementations because they are not expecting to find more than one buildpack on a blob layer.
+
+
+<!--
+This section should document breaks to public API and breaks in compatibility due to this RFC's proposed changes. In addition, it should document the proposed steps that one would need to take to work through these changes. Care should be give to include all applicable personas, such as platform developers, buildpack developers, buildpack users and consumers of buildpack images.
+-->
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we *not* do this?
+
+Distributing a buildpackage with more than one buildpack in a layer blob adds complexity to platform implementors because they will need to do the opposite process when consuming the OCI images.
+
+It will add complexity to Buildpack Authors processes because they will need to think how to group their buildpacks together in a layer blob in an efficient way avoiding for example network issues when pulling new blobs from a registry. Size, frequency of change and other variables must be taken into consideration when buildpacks are flatten.
+
+It will create artifacts that are not consumable by older platforms.
+
+
+# Alternatives
+[alternatives]: #alternatives
+
+- What other designs have been considered?
+
+Some other alternatives mentioned are: squashing by the buildpack size or squashing a CNB Builder when the number of layers is reaching the limit, but those ideas, do not deal with the main problems of distributing more than one buildpack in a layer blob.
+
+
+- Why is this proposal the best?
+
+Not sure if it is the best, but way to solve the `layer limit error` is to optimize the uses of the layer in a OCI image.
+
+- What is the impact of not doing this?
+
+Buildpack Authors and Platform Operators will keep seeing the layer limit error. 
+
+# Prior Art
+[prior-art]: #prior-art
+
+Discuss prior art, both the good and bad.
+
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+
+- What parts of the design do you expect to be resolved before this gets merged?
+
+We think the `--depth` command MAY not be helpful, after implementing a PoC. Should we get rid of this flag?
```

Comment:
Actually, same question for `--flatten-exclude`. Do we need it?

Reply:
Given that Paketo has started to use this argument in a pre-release version of pack, could I ask that if you intend to remove exclude, that you phase it out over a release or two, to give us time to migrate to `--flatten-include`